### ISSUE TITLE
About retrain result

### ISSUE THREAD

**Issue by Endless-Hao** [cardwing/Codes-for-Lane-Detection#23]
@cardwing  hello, I am coming again. I have retrained your pre-train-model in CULane data. In the training process, I think the train is normal. Like accuracy is 70% to  80 %, accuracy-back is 90% to 98%. The times of train epoch is 90000. They are normal.
But when I test them, I also get the not good result. I evaluate the results. Get the information below:
-----------------------------------
Evaluating the results...
tp: 6783 FP: 96950 fn: 98103
finished process file
precision: 0.065389
recall: 0.0646702
F-measure: 0.0650276
----------------------------------
The result in real pictures are like below:

![Uploading 00900.results.png…]()
![Uploading 02430.results.png…]()
![03240 results](https://user-images.githubusercontent.com/19277849/50192807-d832bb00-036e-11e9-9d63-45a8d0f54e60.png)

Is there anything wrong in test code? I retrain it,  the details are normal. But the test results were not good.

**Comment by Endless-Hao** [comment]
@cardwing  Also different test processes get different results using the same model. I really don't know what's wrong with it. I retrain normal.

**Comment by Endless-Hao** [comment]
@cardwing  I upload the full code ，Thank you for your kind.
[Fullcode.zip](https://github.com/cardwing/Codes-for-Lane-Detection/files/2693165/Fullcode.zip)

**Comment by cardwing** [comment]
@Endless-Hao, can you upload your trained model here so that I can have a test on my local server?

**Comment by Endless-Hao** [comment]
@cardwing  Here is my trained model. At the last the accuracy can also achieve 90% or more,  I just say they are in this interval. I upload in google cloud.
https://drive.google.com/open?id=1fLxMfgEpLXQqCl2SENsuB3nub4JW_IVz

**Comment by Endless-Hao** [comment]
@cardwing I also get accuracy above 90% and get accuracy-back above 97% at last. The training process is ok and normal which is nearly the same as the trained picture you upload in another issue.

**Comment by cardwing** [comment]
@Endless-Hao, it is obvious that the problem exists in your evaluation process. The following figure is obtained via your uploaded model. It looks fine and the overall F1-measure is also normal. The following log is obtained via your model. Please carefully check your evaluation code (scripts provided by [SCNN](https://github.com/XingangPan/SCNN)). Besides, you should train the model from the provided vgg.npy instead of the final testing model. 

![test_img](https://user-images.githubusercontent.com/23656119/50200857-b814f300-0392-11e9-9781-1e60a7870661.png)

**Comment by cardwing** [comment]
[vgg_SCNN_DULR_w9_iou0.5.txt](https://github.com/cardwing/Codes-for-Lane-Detection/files/2693432/vgg_SCNN_DULR_w9_iou0.5.txt)

[vgg_SCNN_DULR_w9_iou0.5_split.txt](https://github.com/cardwing/Codes-for-Lane-Detection/files/2693433/vgg_SCNN_DULR_w9_iou0.5_split.txt)

**Comment by Endless-Hao** [comment]
@cardwing  With this different test result, all I can think of is that some of the parameters are from the model, and some of them are randomly initialized, otherwise it is impossible.

**Comment by cardwing** [comment]
@Endless-Hao, your provided probability maps are indeed unsatisfactory. The testing code looks fine as it is exactly the same as mine. The mismatch problem can not be caused by the random initialization of some model parameters. What is the version of your tensorflow?

**Comment by cardwing** [comment]
And it is also weird that the code cannot be debugged in your pycharm.

**Comment by Endless-Hao** [comment]
@cardwing  the version of my tensorflow is 1.10.1.

**Comment by cardwing** [comment]
@Endless-Hao, the version of my tensorflow is 1.3.0. Maybe some functions have been changed as tensorflow is updated. You need to check the problem by your self since I do not have any idea about what to do next.

**Comment by cardwing** [comment]
Another cause why you have different outputs is that BN parameters may not be fixed in the testing phase.

**Comment by Endless-Hao** [comment]
@cardwing  I change another computer which the version of tensorflow is 1.7. And then, I get the good results.
![05130 results](https://user-images.githubusercontent.com/19277849/50212941-37b4b900-03b7-11e9-8798-eb3bf870242e.png)
![05250 results](https://user-images.githubusercontent.com/19277849/50212957-3daa9a00-03b7-11e9-801e-ef9c55bf584b.png)
Maybe is the problem of some function in tensorflow. Thank you for your help very much.

**Comment by cardwing** [comment]
^_^

**Comment by ktr-hubrt** [comment]
> ^_^

first of all, thank for your amazing work.
when i try to follow your work，during training， i found the training accuracy is up to 90%，so i want to know when the model is good enough?

**Comment by cardwing** [comment]
The training accuracy is fine. Just try the evaluation process and see if you can achieve similar performance.

**Comment by ktr-hubrt** [comment]
> The training accuracy is fine. Just try the evaluation process and see if you can achieve similar performance.

3q so much

**Comment by ktr-hubrt** [comment]
> The training accuracy is fine. Just try the evaluation process and see if you can achieve similar performance.

After the model i trained get the f1score 0.68, which is 3 point lower than your model, i wonder how you trained your model, the same parameters as the global_config?
